addons/website_slides/models/slide_channel.py

```diff
@@ -670,3 +675,49 @@ def _resequence_slides(self, slide, force_category=False):
 
     def get_backend_menu_id(self):
         return self.env.ref('website_slides.website_slides_menu_root').id
+
+    def request_access_to_responsible(self):
```

Comment:
I think we can split this method in two

  * a private method _action_request_access, doing the core computation (fetch channels to join, create activities), and that returns the new activities;
  * a public one, returning a structured result ('error' if already member (can be done before calling _action_request_access, preventing public users from calling it, ...);

That way we separate business code (private method creating records) from official API returning some structured data.